### ISSUE TITLE
How can i use this repo to add sunglasses like other attributes to faces.

### ISSUE THREAD

**Issue by alan-ai-learner** [MartinPernus/MaskFaceGAN#4]
Any suggestion suggestion would be great.
Thanks

**Comment by MartinPernus** [comment]
Unfortunately I have not had any success adding glasses with this method. I found InterFaceGAN better suited for adding eyeglasses, but it comes with several drawbacks that were mentioned in the paper. Another option is the mapping method of StyleCLIP - their embedding method is similar to this one, so I presume it would not work.

I have also experimented with adding a specific pair of eyeglasses onto an image, again with trying with many different techniques. The only thing that really worked was projecting 3D model of glasses onto image plane together with the face and then embedding it. Using E4E encoder for initialization, I think you could process a single image in something like a minute if processing on GPU. 

Of course the difficult thing is obtaining 3D models of eyeglasses (or other accessories) - there are some websites that contain free models but it's not exactly an abundance to my knowledge. Which is too bad - this could be very useful for some apps, where you try out different glasses just with your phone!

**Comment by alan-ai-learner** [comment]
Hey thank you so much for replying so quickly, as you are saying it seems
like you have already done a lot of research on this so from now on if I
have to add glasses to the face what are the possible ways that can be
explored in this area also I have one question regarding the faceapp We are
using for the adding attribute in to the faces how are they managing to do
it any idea would be great.

**Comment by MartinPernus** [comment]
Hi,

for basic attribute editing, perhaps you could look into AttGAN and STGAN papers. MaskFaceGAN is primarily concerned with high resolution disentangled editing, but is slower than traditional methods. If that is not a concern, those methods could be your best bet.

**Comment by lzaazl** [comment]
Nice work!

Could you please share the attribute classifier training code? Thanks in advance.

**Comment by RuoyuChen10** [comment]
> Nice work!
> 
> Could you please share the attribute classifier training code? Thanks in advance.

You can reference this repo to train attribute net.

[https://github.com/RuoyuChen10/Multi-label-on-VOC2008-attributes](https://github.com/RuoyuChen10/Multi-label-on-VOC2008-attributes)
